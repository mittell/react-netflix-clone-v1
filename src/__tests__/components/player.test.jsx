/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/prefer-presence-queries */
/* eslint-disable testing-library/prefer-screen-queries */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Player } from '../../components';

describe('<Player />', () => {
	it('renders the <Player/> with a video', () => {
		const { container, queryByTestId, getByText } = render(
			<Player>
				<Player.Button />
				<Player.Video src='/videos/bunny.mp4' />
			</Player>
		);

		// No Player and clicks to Open
		expect(queryByTestId('player')).toBeFalsy();
		fireEvent.click(getByText('Play'));

		// Player is visible and clicks to Close
		expect(queryByTestId('player')).toBeTruthy();
		fireEvent.click(queryByTestId('player'));

		// Player is closed
		expect(queryByTestId('player')).toBeFalsy();
		expect(container.firstChild).toMatchSnapshot();
	});
});
