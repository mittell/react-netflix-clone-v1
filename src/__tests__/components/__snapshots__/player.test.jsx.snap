// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> renders the <Player/> with a video 1`] = `
<div
  class="player__Container-sc-16zk2s6-0"
>
  <button
    class="player__Button-sc-16zk2s6-3 cDKZDU"
  >
    Play
  </button>
</div>
`;
